File: Coinster/Model/Coin.swift
//
//  Coin.swift
//  Coinster
//
//  Created by Christopher Lee on 9/12/17.
//  Copyright © 2017 Christopher Lee. All rights reserved.
//

import Foundation

struct Coin: Codable {
    let id: String
    let name: String
    let symbol: String
    let priceUSD: String
    let priceBTC: String
    let marketCapUSD: String?
    let percentChange24h: String?
    
    var imageUrl: String {
        return "https://files.coinmarketcap.com/static/img/coins/32x32/\(id).png"
    }
    
    private enum CodingKeys: String, CodingKey {
        case id = "id"
        case name = "name"
        case symbol = "symbol"
        case priceUSD = "price_usd"
        case priceBTC = "price_btc"
        case marketCapUSD = "market_cap_usd"
        case percentChange24h = "percent_change_24h"
        
    }
    
//    init(id: String, name: String, symbol: String, price_usd: String, price_btc: String, market_cap_usd: String?, percent_change_24h: String?) {
//        self.id = id
//        self.name = name
//        self.symbol = symbol
//        self.price_usd = price_usd
//        self.price_btc = price_btc
//        self.market_cap_usd = market_cap_usd
//        self.percent_change_24h = percent_change_24h
//    }
}

File: coinsterAPI/api/models/coin.js
/** coin.js **/

var Coin = function (data) {
  this.id = data.id;
  this.name = data.name;
  this.symbol = data.symbol;
  this.price_btc = data.price_btc;
  this.price_usd = data.price_usd;
  this.market_cap_usd = data.market_cap_usd;
  this.last_updated = data.last_updated;
  this.total_supply = data.total_supply;
  this.percent_change_1h = data.percent_change_1h;
  this.percent_change_24h = data.percent_change_24h;
  this.percent_change_7d = data.percent_change_7d;
}

Coin.prototype.data = {}

module.exports = Coin;